---
title: "Count_matrix_analyses"
author: "LounaCS"
date: "12/15/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Import the data with tximport

## build metadata (link between biological annotations and analyses annotations)


```{r}
library(tidyverse)

sampleTable = read_delim("metadata_samples.tsv") %>% as.data.frame()
rownames(sampleTable) = sampleTable$sample_ID
sampleTable
```

##  create the count table with all samples using tximport

```{r}
library(tximport)

dir <- "/home/rstudio/mydatalocal/TP_elimination_L/3_processed_data/quantification/"
samples <- list.dirs(dir, full.names = F) 
samples <- samples[samples != ""] # remove "." directory

samples_names <- rownames(sampleTable)

files <- file.path(dir, samples_names, "abundance.h5")
names(files) <- samples_names
files[!file.exists(files)]


txi.kallisto.tmp <- tximport(files[1], type = "kallisto", txOut = TRUE)

transcript_names <- rownames(txi.kallisto.tmp$abundance)
tx2gene <- data.frame(tx=transcript_names,
                  gene=gsub(".t[0-9]*","",transcript_names))
txi.kallisto <- tximport(files, type = "kallisto", tx2gene = tx2gene)
```





# Differential expression (DE) analyses using DESeq2

## build the DESeq2 object

```{r}
library(DESeq2)

sampleTable$during = "not during"
sampleTable$during[sampleTable$timing == "during"] = "during"

dds <- DESeqDataSetFromTximport(txi.kallisto, sampleTable, ~ elimination )
dds <- DESeq(dds)
#DESeq compares statistically genetic expression under different hypothetic factors. also normalize the count matrix.
vsd <- vst(dds, blind=FALSE)
```

## ACP of gene differential expression
```{r}
plotPCA(vsd, intgroup=c("age"))
```
 The first principal component (PC) seems to differenciate samples according to the corresponding developmental stade. It shows that the higher data dispersion is due to developmental processes, which is expected.
 Replicates are aggregated on the PCA, which highlight the similitudes between samples of the same age.
 The second principal component seems to discriminate samples according to the genome elimination event: samples from before and after the genome elimination have low PC2 values, while samples from the stades 4- and 8-cells have higher PC2 values. That segregation indicates a differencial expression linked to the genome elimination event.


## Count matrix
The count matrix is a large matrix with p-value (the robustness of differencial expression between 'during' elimination and 'not during', meaning before and after genome elimination) and Log2FoldChange (the intensity of differencial expression between the two conditions).

```{r}
res = results(dds)

res_df <- as.data.frame(res)
head(res_df)
```

What is the number of significantly up/downregulated genes?
```{r}
summary(res)
```
We obtain 1163 gene upregulated.

To identify them, we can:
 - plot their expression profile
 - identify their category

```{r}
plotCounts(dds, gene="mbelari.g8166", intgroup = "age")
```
According to their expression profile, upregulated genes are mainly highly expressed after the genome elimination.
Significatively downregulated genes are rather expressed in late stades.
Those patterns reflects developmental processes, thus are of little interest for us.

We can adjust the threshold of significance to study genes with lower p-value of differential expression.


### Plot orientation
```{r}
resultsNames(dds)
```
### Plot of the down/upregulated genes

```{r}
plotMA(res, ylim=c(-30,30))
```
 
 
 
# Selection of genes of interest for the biological question
 We can't identify all upregulated genes, so we focuss on the most upregulated genes.

## Useful functions
### Get annotation data from wormbase parasite

Those annotations come from the C. elegans database. All belari genes are not annotated as this species is not a model one.
The biomaRt library allows us to get data from the Wormbase ParaSite website.

```{r}
# get annotations from database

# BiocManager::install("biomaRt") # only one time

library(biomaRt)
mart <- useMart("parasite_mart", dataset = "wbps_gene", host = "https://parasite.wormbase.org", port = 443)

genes_annotation <- getBM(mart = mart,
                          filters = c("species_id_1010"),
                          value = list("mebelaprjeb30104"),
                          attributes = c("wbps_gene_id",
                                         "caelegprjna13758_gene_name",
                                         "caelegprjna13758_homolog_ensembl_peptide",
                                         "interpro_id", #conserved domains across proteins
                                         "interpro_short_description")
)


```

### Creating graphs

The subset method applies filters on columns.
To create the graphs of expression patterns for various genes of interest, we can use this function:

```{r}

# function to create graphs
plot_counts <- function(X) {
count_df <- counts(dds, normalized=T) #dds: count matrix
count_dfl = count_df %>% as.data.frame() %>%  rownames_to_column("gene_ID") %>% pivot_longer(-gene_ID,names_to = "sample_ID", values_to = "norm_counts")


df_tmp = subset(count_dfl, gene_ID %in% X) %>% left_join(sampleTable)

df_tmp <- df_tmp %>% mutate(age = factor(age, levels = unique(age)),
                            gene_ID = factor(gene_ID,levels=X))


ggplot(df_tmp, aes(x=age, y = norm_counts, col = age)) + theme_bw() +
  geom_point() +
  facet_wrap(~gene_ID, scales = "free_y" ) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
}
```

# Analysis: Gene identification
## First question_Selection of genes highly up/downregulated during the genome elimination phase

Here, we wonder **what genes are the most upregulated genes during genome elimination?**

We can adjust the number of significantly upregulated genes with the parameter log2FoldChange (maximum strength:20).
 
```{r}
upregulated = subset(res_df, log2FoldChange > 4 & baseMean > 10 & padj < 0.05) #noms de genes dans les noms de ligne: transfÃ©rer dans une colone pour lecture facile
downregulated = subset(res_df, log2FoldChange < -7 & baseMean > 10 & padj < 0.05)
```

```{r}
summary(upregulated)
```

### creating graphs
```{r}
upregulated$gene_ID

plot_counts(upregulated$gene_ID)
```
We see that upregulated genes are mainly upregulated during the late phase of development. It is thus not really interesting for us.

```{r}
downregulated$gene_ID_down
plot_counts(downregulated$gene_ID)
```
We see that downregulated genes exhibit various patterns, that may be due to the consumption of maternal RNA before the transcription activation.

## Second question: robustly regulated genes among cells in the samples, i.e. with the higher p-value.

We here wonder **what function display genes that are most robustly regulated?**


To do so, we first sort genes whose regulation is significant by decroissant p-value, then look at their functions.
(de: differential expression)

```{r}
# adjusted parameters: log2FoldChange=y-axis, baseMean=x-axis,n padj=p-adjusted value, 

res_df  <- res %>% as.data.frame() %>% rownames_to_column("gene_ID")
genes_list <- subset(res_df, log2FoldChange < -5 & padj <0.05) %>% arrange(padj) %>% head(20)
plot_counts(genes_list$gene_ID)
robust <- subset(genes_annotation, wbps_gene_id %in% genes_list$gene_ID) # we select annotations for genes of interest from the total annotation table

head(robust)
```


```{r}
genes_annotation[10:20,]
```

## Third question: are there genes with interesting functions in the filtered genes?

We can mine our dataset of filtered genes by keywords with the grep method, looking for proteic functions of interest: exonuclease, transposase, or helicase.

```{r}
filtered_gene = upregulated # OR robust OR downregulated
head(filtered_gene)

toMatch = c("exonuclease", "transposase", "helicase")
exotranspoheli = filtered_gene[grep(paste(toMatch,collapse="|"),
                                              filtered_gene$interpro_short_description, ignore.case = T),]
exotranspoheli

plot_counts(unique(exotranspoheli$wbps_gene_id))
```

We can also look for helicases.
